corehq/pillows/case_search.py

```diff
@@ -94,12 +89,9 @@ def _add_smart_types(dynamic_properties, domain, case_type):
     # Properties are stored in a dict like {"key": "dob", "value": "1900-01-01"}
     # `value` is a multi-field property that duck types numeric and date values
     # We can't do that for properties like geo_points in ES v2, as `ignore_malformed` is broken
-    if USH_CASE_CLAIM_UPDATES.enabled(domain):
-        gps_props = get_gps_properties(domain, case_type)
-        _add_gps_smart_types(dynamic_properties, gps_props)
-    if GEOSPATIAL.enabled(domain):
-        gps_props = [get_geo_case_property(domain)]
-        _add_gps_smart_types(dynamic_properties, gps_props)
+    gps_props = get_gps_properties(domain, case_type)
```

Comment:
This is the only part that gives me any pause from a performance/scale perspective.  It does a DB lookup, though it is cached for 24 hours.  It's a shame that it's probably just `[]` for the overwhelming majority of domains and case types, but I can't think of any great alternatives.

Could probably instead store a cached dict mapping `(domain, case_type)` to the list of properties for only those combinations where there _are_ gps properties, so it'd only be one stored cache value, but that would perform poorly if in the future we do see high usage rates.  I dunno, probably fine as is, just wanted to make sure we're thinking about it.

Reply:
Hey @esoergel, thanks for taking a think through this. Thinking through the alternative you suggested I agree doing a single cache across all domains might have some performance implications at scale.

I also considered caching based on domain only. We would essentially do a cached dict of GPS case properties for all case types in the domain. So if a domain is doing frequent form submissions for lots of different case types we could simply cache all of them at once. I'm not sure if this is a worthwile alternative however, since this can cause needless caching of GPS case properties which (almost) never get used.

A quick win here could be to simply add a privilege check for the `DATA_DICTIONARY` so that we only do a DB lookup for domains that have the privilege. For domains that don't, we simply return `[]`. At the very least, this can help cut back on DB lookups for domains that have an empty DD or don't use it at all. There might be domains that lose access to the DD privilege but still have GPS case properties, but since these domains will be unable to update/change their GPS case properties, I feel we shouldn't still attempt to process these properties.

Reply:
Yeah, no strong preference on my end - also fine to leave as-is for now and revisit later if needed.

Reply:
> A quick win here could be to simply add a privilege check for the DATA_DICTIONARY so that we only do a DB lookup for domains that have the privilege

I like this, and I strongly recommend that this should be added.

Reply:
Is same true for `get_geo_case_property`? Is that relevant only if `GEOSPATIAL` is enabled?

Reply:
>  I like this, and I strongly recommend that this should be added.

I agree. Addressed in [7a84adb](https://github.com/dimagi/commcare-hq/pull/35001/commits/7a84adb1108d5306c6c18f7b31d7b1f2b30ec83a).

> Is same true for get_geo_case_property? Is that relevant only if GEOSPATIAL is enabled?

If the domain doesn't have the `GEOSPATIAL` feature flag enabled, then `get_geo_case_property()` will simply return a default `gps_point` case property name. This is getting saved to ES irrespective of whether the feature flag is enabled so that, even if the domain only enables the flag at a later point, their GPS data will already be populated in ES.

Reply:
Cool, and this is to maintain historic data before the feature flag is enabled right?
I guess that does not need a comment in code so all should be good here

Reply:
That is correct, this wiil help make scenarios where we have to run a management command to bring a domain's data up-to-date less frequent.